tools/nodejs/lib/duckdb.d.ts

```diff
@@ -4,12 +4,54 @@
  * on Node.JS API
  */
 
+export type ExceptionType =
+    | "Invalid"          // invalid type
+    | "Out of Range"     // value out of range error
+    | "Conversion"       // conversion/casting error
+    | "Unknown Type"     // unknown type
+    | "Decimal"          // decimal related
+    | "Mismatch Type"    // type mismatch
+    | "Divide by Zero"   // divide by 0
+    | "Object Size"      // object size exceeded
+    | "Invalid type"     // incompatible for operation
+    | "Serialization"    // serialization
+    | "TransactionContext" // transaction management
+    | "Not implemented"  // method not implemented
+    | "Expression"       // expression parsing
+    | "Catalog"          // catalog related
+    | "Parser"           // parser related
+    | "Binder"           // binder related
+    | "Planner"          // planner related
+    | "Scheduler"        // scheduler related
+    | "Executor"         // executor related
+    | "Constraint"       // constraint related
+    | "Index"            // index related
+    | "Stat"             // stat related
+    | "Connection"       // connection related
+    | "Syntax"           // syntax related
+    | "Settings"         // settings related
+    | "Optimizer"        // optimizer related
+    | "NullPointer"      // nullptr exception
+    | "IO"               // IO exception
+    | "INTERRUPT"        // interrupt
+    | "FATAL"            // Fatal exceptions are non-recoverable and render the entire DB in an unusable state
+    | "INTERNAL"         // Internal exceptions indicate something went wrong internally (i.e. bug in the code base)
+    | "Invalid Input"    // Input or arguments error
+    | "Out of Memory"    // out of memory
+    | "Permission"       // insufficient permissions
+    | "Parameter Not Resolved" // parameter types could not be resolved
+    | "Parameter Not Allowed"  // parameter types not allowed
+    | "Dependency"       // dependency
+    | "Unknown"
+    ;
+
 /**
  * Standard error shape for DuckDB errors
  */
 export interface DuckDbError extends Error {

```

Comment:
There's another option where you can set the `DuckDbError` as a `class` then users could use the `new` keyword with `DuckDbError`
```ts
class DuckDBErr extends Error {
  public errno: number = -1;
  public code: string = 'DUCKDB_NODEJS_ERROR';
  public errorType : ExceptionType;
  constructor(message: string, errorType : ExceptionType) {
    super(message);
    this.errorType = errorType
   // ...
  }
}
```

Then user could do

```ts

const myError : DuckDBErr = new DuckDBErr("message", 'FATAL')

```

Also, another approach the type `ExceptionType` is to have it as `enum` similar to:

```ts

export enum duckException {
  FATAL='FATAL'
}

```
Then user could do

```ts

const myError : DuckDBErr = new DuckDBErr("message", duckException.FATAL)

```

Reply:
I'm not sure how making it a class instead of an interface adds much, to be honest? I don't see users constructing instances themselves. And as far as enum vs union type goes, enums wind up being more verbose, and less succinct in terms of the algebraic types I'll be adding to this for the http exception shortly